src/main/java/com/teragrep/pth_06/planner/walker/conditions/LatestCondition.java

```diff
@@ -84,13 +84,25 @@ public Condition condition() {
 
     @Override
     public boolean equals(final Object object) {
-        if (this == object)
+        if (this == object) {
             return true;
-        if (object == null)
+        }
+        if (object == null) {
             return false;
-        if (object.getClass() != this.getClass())
+        }
+        if (object.getClass() != this.getClass()) {
             return false;
+        }
         final LatestCondition cast = (LatestCondition) object;
         return this.value.equals(cast.value);
     }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
```

Comment:
hashcode

Reply:
Solved in commit 4d4e0023426356b05c592147f65933f69ba5dfa3